code/modules/mob/living/simple_animal/hostile/megafauna/onestarboss.dm

```diff
@@ -0,0 +1,178 @@
+/mob/living/simple_animal/hostile/megafauna/one_star
+	name = "Type - 0315"
+	desc = "Love and concrete."
+
+	faction = "hive"
+
+	icon = 'icons/mob/64x64.dmi'
+	icon_state = "onestar_boss_unpowered"
+	icon_living = "onestar_boss_unpowered"
+	icon_dead = "onestar_boss_wrecked"
+	pixel_x = -16
+	ranged = TRUE
+
+	health = 1700
+	maxHealth = 1700
+	break_stuff_probability = 95
+	stop_automated_movement = 1
+
+	melee_damage_lower = 10
+	melee_damage_upper = 20
+	megafauna_min_cooldown = 30
+	megafauna_max_cooldown = 60
+
+/mob/living/simple_animal/hostile/megafauna/one_star/death()
+	..()
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/LoseTarget()
+	stance = HOSTILE_STANCE_IDLE
+	target_mob = null
+	icon_state = initial(icon_state)
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/LostTarget()
+	stance = HOSTILE_STANCE_IDLE
+	icon_state = initial(icon_state)
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/FindTarget()
+
+	var/atom/T = null
```

Comment:
This whole proc is a copypaste of hostile animal one's too, with two additions: light check to change range of search and icon updating.

The light check may be worth moving to parent proc, so all animals can be evaded in total darkness.
If that's undesirable, should add `/mob/living/simple_animal/hostile/proc/get_view_range()` which just returns 10, and override here to return 10 or 4 depending on light.

After that this proc could be just
```
/mob/living/simple_animal/hostile/megafauna/one_star/FindTarget()
  . = ..()
  if(.)
     update_icon() (this 'if no target use one icon, if target use another icon' thing should really be moved to update_icon cause it's done in 3 places already)
```